geth/peers/topicpool.go

```diff
@@ -240,38 +232,40 @@ func (t *TopicPool) StartSearch(server *p2p.Server) error {
 	if server.DiscV5 == nil {
 		return ErrDiscv5NotRunning
 	}
+	atomic.StoreInt32(&t.running, 1)
+
 	t.mu.Lock()
 	defer t.mu.Unlock()
-	atomic.StoreInt32(&t.running, 1)
+
 	t.quit = make(chan struct{})
-	t.period = make(chan time.Duration, 2) // 2 allows to send slow and then fast without blocking a producer
-	found := make(chan *discv5.Node, 5)    // 5 reasonable number for concurrently found nodes
-	lookup := make(chan bool, 10)          // sufficiently buffered channel, just prevents blocking because of lookup
+	found := make(chan *discv5.Node, 5) // 5 reasonable number for concurrently found nodes
+	lookup := make(chan bool, 10)       // sufficiently buffered channel, just prevents blocking because of lookup
+
+	// start syncing mode strategy
+	period := t.sync.Start()
+
 	if t.cache != nil {
 		for _, peer := range t.cache.GetPeersRange(t.topic, 5) {
 			log.Debug("adding a peer from cache", "peer", peer)
 			found <- peer
 		}
 	}
-	t.discWG.Add(1)
+
+	t.wg.Add(1)
 	go func() {
-		server.DiscV5.SearchTopic(t.topic, t.period, found, lookup)
-		t.discWG.Done()
+		server.DiscV5.SearchTopic(t.topic, period, found, lookup)
+		t.wg.Done()
 	}()
-	t.consumerWG.Add(1)
+	t.wg.Add(1)
```

Comment:
without mutliple groups you can get read from closed channel

Reply:
You mean multiple `TopicPools`? I was trying to figure out why it is so and could not reproduce any issues after changing.

Reply:
ok it seems not possible now

previously i could close t.period when t.sync is stopped
but handleServers was still running and could write to period